packages/cli-repl/src/cli-repl.ts

```diff
@@ -62,16 +62,43 @@ class CliRepl {
       this.connect(driverUri, driverOptions);
     }
   }
+  /**
+   * Returns true if a value is a shell api type
+   *
+   * @param {any} evaluationResult - The result of evaluation
+   */
+  private isShellApiType(evaluationResult: any): boolean {
+    return evaluationResult &&
+      typeof evaluationResult.shellApiType === 'function' &&
+      typeof evaluationResult.toReplString === 'function'
+  }
 
   /**
-   * The custom evaluation function.
+   * The custom evaluation function. Evaluates the provided input and format it
+   * for the output.
    *
    * @param {} originalEval - The original eval function.
    * @param {} input - The input.
    * @param {} context - The context.
    * @param {} filename - The filename.
    */
-  async evaluator(originalEval: any, input: string, context: any, filename: string) {
+  async evaluateAndFormat(originalEval: any, input: string, context: any, filename: string) {
+    const evaluationResult = await this.evaluate(originalEval, input, context, filename);
+
+    if (this.isShellApiType(evaluationResult)) {
+      return formatOutput(
+        await evaluationResult.toReplString(),
+        evaluationResult.shellApiType()
+      );
+    }
+
+    return formatOutput(evaluationResult);
```

Comment:
i think the most proper way of doing formatting in the cli is through repl's `writer` function. It's one the [options](https://nodejs.org/api/repl.html#repl_repl_start_options) we pass in to repl.start.

My reasoning there is that I want to use as much of the formatting as node repl provides us and as much of the standard API as possible.

Reply:
I think you're right. Don't know why i ended up keeping the one in evaluate instead of putting it there.